packages/x-charts/tsconfig.build.json

```diff
@@ -8,9 +8,16 @@
     "noEmit": false,
     "emitDeclarationOnly": true,
     "outDir": "build",
-    "rootDir": "./src"
+    "rootDir": "./src",
+    "types": [
```

Comment:
Yes, this is great! Is there a reason to not place it in `tsconfig.json`? I feel like it would be beneficial to keep the config for typechecking as close as possible to the one for type generation. This file should adopt the setting automatically through the `extends`.

Reply:
I never really understood why we have a separate tsconfig for the build.
@LukasTy @Janpot Can you enlighten me, please? 🙂

Reply:
I suspect one of two things
1. for typechecking we use `noEmit: true` and for type generation we use `noEmit: false`
2. for type generation we're working towards a setup with a single compilation using project references instead of a compilation per project.

but @michaldudak will be able to shed more light

Reply:
> But I hope we can bring this setup to type checking as well.

This is already set up like this in the Base UI repo. However, it doesn't work perfectly (yet), as it doesn't typecheck tests.

> for typechecking we use noEmit: true and for type generation we use noEmit: false. But I believe this should be achievable through a cli flag as well.

True. I believe all settings are available through commandline flag. It's a matter of readability to create a config file.

In addition to what @Janpot wrote, the build config also excludes tests from the output.